### DISCUSSION TITLE
Increased Ping time following Jamulus upgrade?

### DISCUSSION THREAD

**Discussion by TPennick** [jamulussoftware/jamulus#1328] [Technical Support & Help]
Hi All,
Apologies if this is a read herring, but this issue occurred during testing following upgrade of my private server, and my Windows client.  
I have been running a private server on a Linode virtual server in London for the past few weeks.  Until now the Ping time has been approximately 6ms (I'm in Ipswich UK).  Following upgrade of my Jamulus server and client, I checked the ping time to find that it is now 23ms.  Wi-Fi is switched off.  
Admittedly, this may be nothing to do with Jamulus, as I have tried pinging from the command line, to my server with the same results, and also looking at the server lists on Jamulus, I find that there is nothing listed under 20ms, whereas there were formerly lots including the Folk server in Frankfurt.
I suspect a problem with my router, a BT HomeHub6, and have re-started this a number of times with no improvement.
Can anyone suggest why upgrading Jamulus might have had this effect on outward connections from my home network, or take pity on me even if this isn't a Jamulus issue and suggest some other reason why the ping times should suddenly have increased in this way (with possible options to address them).
Regards and thanks,
Tim Pennick

**Comment by hoffie** [comment]
Hi,

there should be no changes in 3.7.0 which would lead to higher ping times. Have you tried downgrading to cross-check?

Depending on the area of living, 20ms is already pretty ok (could be much worse). Although Jamulus does not use ICMP pings, you could try a traceroute to see where ping times are increasing.

Kind regards,
Christian

**Reply by ghost** [reply]
Do you run Jamulus with/out mulithreading? (eg. Jamulus -s -T)
I have a counterintuitive situation where my single core (IA32 core2) computer has worse ping times with -T.
So check your settings also.

**Comment by TPennick** [comment]
Thanks Christian.  Traceroute was a great idea I hadn’t thought of, and identifies the issue as arising between my home router, and the next router upstream which is shown as having 20ms delay.  All other steps don’t add more than a couple of ms.  I realise this is questionably off topic for this discussion as I believe now that the issue isn’t caused by Jamulus, however Jamulus experts are likely to have encountered these issues and might be in a position to advise as to what I could do next.  It’s annoying me, because I know it can be better than it is currently, as up until the past couple of days it was only showing 5 or 6 ms between me and my server in London.

 

Regards and thanks for any thoughts,

 

Tim

**Reply by ve3meo** [reply]
Stuff happens. The Internet is a mesh of communications paths with routers at each intersection looking for the next easy one to pass your data packets. Two Jamulus servers on the same cloud host, therefore in the same building, maybe even in the same rack, maybe even on the same computer. Identical or near identical ping times for weeks. Suddenly one (mine) jumps by several ms while the other remains the same. TraceRoute showed that mine was 12 hops via the Cogent network while the other was 9 hops via the GTT net. Some days passed and mine briefly returned to the shorter ping, fewer hops, GTT net. The next day it bounced up again and was back 12 on Cogent. Where's the cause? My ISP? Some intervening router? Or the host?

**Reply by gene96817** [reply]
Which network is your client connected to? What are the details of the communication of the hosting contract between the two cloud servers? Your ISP's router (really its the routing table) don't see the two servers in the same way. If the two servers were identical targets then it would be random as to which server gets the longer path. 

It also matters what peering contracts your ISP has with other ISPs. Here is a made up example to illustrate the point. You are a subscriber of ISP A. A has more bandwidth connection to Cogent and much less bandwidth with GTT.  On some days the A-to-GTT connection is congested and the router sends your traffic to Cogent. On those days, it is possible that the ping times through GTT could be longer than through Cogent. In this case, the root cause could be another subscriber's server loading a link on GTT that causes the routing decision to change.

Yeah, stuff happens. If you really want to know, save the traceroute paths. Ping the routers on the 9 hop path. When your traffic is switched to the 12 hop path, ping the routers on the 9 hop path. You might see some different ping times. This is not a perfect test but it might give you a clue.

**Comment by TPennick** [comment]
To return to the original issue, I thought people might be interested in what I think is the solution.  Tracert showed that the issue of the additional 20ms was arising on the first hop after my personal router.  My line is a 'fibre to the cabinet' BT line, so copper from the cabinet to my master socket.  an engineer visit confirmed that there were lots of faults on the line, which were fixed by the engineer, however the extra 20ms hop time was still present.  Reading around, I learn that there is a method of counteracting problems with noisy fibre lines called 'interleaving' (though my terminology may be incorrect and I can't claim to be an expert), which I believe had been implemented on my line as a result of the faults subsequently cleared by the engineer.  Apparently, it is possible for Openreach, the arm of BT which manages the local network to re-set the system so that Interleaving is removed.  I was considering requesting this, but looking at my Tracert results today suggests that the ping time is dropping, so I plan to leave well alone and see how the system supports Jamulus in the next few days.
Regards and thanks to all for suggestions.

**Reply by gene96817** [reply]
You are very lucky that BT was willing to explain their side of the router configuration.
